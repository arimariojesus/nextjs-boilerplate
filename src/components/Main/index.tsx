const Main = () => (
  <div>
    <h1>Hello World</h1>
  </div>
);

export default Main;
