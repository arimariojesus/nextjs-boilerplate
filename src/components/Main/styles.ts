import styled from 'styled-components';

export const MainContainer = styled.div`
  width: 100%;
  height: 100%;
  background-color: #0071ff33;

  display: flex;
  align-items: center;
  justify-content: center;
`;
