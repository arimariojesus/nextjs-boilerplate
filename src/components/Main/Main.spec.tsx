import { render, screen } from '@testing-library/react';

import Main from '.';

describe('<Main />', () => {
  it('should render hello world title', () => {
    const { container } = render(<Main />);

    expect(
      screen.getByRole('heading', { name: /hello world/i })
    ).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });
});
