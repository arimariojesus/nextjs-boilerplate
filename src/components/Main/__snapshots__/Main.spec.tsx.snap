// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should render hello world title 1`] = `
.c0 {
  width: 100%;
  height: 100%;
  background-color: #0071ff33;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

<div
  class="c0"
>
  <h1>
    Hello World
  </h1>
</div>
`;
